File: examples/java/src/main/java/io/github/json2varssetter/JsonParser.java
package io.github.json2varssetter;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.File;
import java.io.IOException;

/**
 * Minimal JSON configuration parser used to demonstrate consuming the
 * json2vars-setter matrix file in a Java project.
 */
public final class JsonParser {

    private JsonParser() {
    }

    /**
     * Parse a JSON configuration file into a JsonNode tree.
     *
     * @param filePath path to the JSON file
     * @return the parsed tree, or {@code null} on failure
     */
    public static JsonNode parseConfig(String filePath) {
        return parseConfig(filePath, false);
    }

    /**
     * Parse a JSON configuration file into a JsonNode tree.
     *
     * @param filePath path to the JSON file
     * @param silent   when true, suppress error output
     * @return the parsed tree, or {@code null} on failure
     */
    public static JsonNode parseConfig(String filePath, boolean silent) {
        try {
            return new ObjectMapper().readTree(new File(filePath));
        } catch (IOException e) {
            if (!silent) {
                System.err.println("Error reading or parsing JSON: " + e.getMessage());
            }
            return null;
        }
    }
}

File: examples/deno/json_parser.ts
// Minimal JSON configuration parser used to demonstrate consuming the
// json2vars-setter matrix file in a Deno project.

export function parseConfig(
  filePath: string | URL,
  silent = false,
): Record<string, unknown> | null {
  try {
    const contents = Deno.readTextFileSync(filePath);
    return JSON.parse(contents) as Record<string, unknown>;
  } catch (e) {
    if (!silent) {
      const message = e instanceof Error ? e.message : String(e);
      console.error(`Error reading or parsing JSON: ${message}`);
    }
    return null;
  }
}

if (import.meta.main) {
  const result = parseConfig(
    new URL("./deno_project_matrix.json", import.meta.url),
  );
  if (result !== null) {
    console.log(JSON.stringify(result, null, 2));
  }
}
